--- lockdev-1.0.1.orig/Makefile
+++ lockdev-1.0.1/Makefile
@@ -11,8 +11,8 @@
 MVER	= ${shell expr `pwd` : '.*-\([0-9]*\).[0-9]*'}
 
 static	= ${libname}.a
-shared	= ${libname}.${VER}.so
-soname	= ${libname}.so.${MVER}
+shared	= ${libname}.so.${VER}
+soname	= ${libname}.so.${VER}
 
 # overwritten by caller (e.g.: debian/rules)
 basedir	= /usr/local
@@ -32,6 +33,7 @@
 
 shared ${shared}:	${objs}
 	${CC} ${LCFLAGS} -shared -Wl,-soname,${soname} $^ -lc -o ${shared}
+	ln -s ${shared} liblockdev.so
 
 
 perl-lib:	static
--- lockdev-1.0.1/LockDev/Makefile.PL.orig	Sun Oct 24 20:10:00 1999
+++ lockdev-1.0.1/LockDev/Makefile.PL	Thu Feb  5 14:17:56 2004
@@ -4,16 +4,7 @@
 WriteMakefile(
          'NAME'	=> 'LockDev',
  'VERSION_FROM'	=> 'LockDev.pm', # finds $VERSION
-         'LIBS'	=> [''],
+         'LIBS'	=> ['-L.. -llockdev'],
        'DEFINE'	=> '',
           'INC'	=> '-I../src',
-     'MYEXTLIB'	=> '../liblockdev.a',
 );
-
-sub MY::postamble {
-'
-$(MYEXTLIB): ../src/lockdev.c ../src/lockdev.h
-	cd .. && $(MAKE) static
-';
-}
-
